core/modules/drr.h

```diff
@@ -0,0 +1,234 @@
+#ifndef BESS_MODULES_DRR_H_
+#define BESS_MODULES_DRR_H_
+
+#include<rte_hash_crc.h>
+
+#include "../module.h"
+#include "../module_msg.pb.h"
+#include "../utils/ip.h"
+#include "../pktbatch.h"
+#include "../kmod/llring.h"
+#include "../utils/cuckoo_map.h"
+
+
+using bess::utils::IPAddress;
+using bess::utils::CIDRNetwork;
+using bess::utils::CuckooMap;
+
+/*
+  * This module implements Discrete Round Robin, a fair queueing algorithm, for flows.
+  * On receiving each packet, they are assigned to a flow's queue based on their 5 tuple.
+  * On running a task, the module goes through the flows in round robin fashion
+  * adding a quantum to their deficit(allocated bytes). The module will then query
+  * each of those flows for packets until their deficit falls below the next packet's
+  * size. After a obtaining a 32 packets(a full batch), the module passes these
+  * packets onto the next module.
+  *
+  * based on this:
+  *  https://en.wikipedia.org/wiki/Deficit_round_robin
+  * EXPECTS: Input packets in any format
+  *
+  * MODIFICATIONS: None
+  *
+  * INPUT GATES: 1
+  *
+  * OUTPUT GATES: 1
+  *
+  * PARAMETERS:
+  *    * quantum: the number of bytes allocated to each flow on each round
+  *    * Max Number of flows: max number of flows the module will handle
+  *    * Max Flow Queue Size: the maximum size that any Flows queue can get
+  *          before the module will start dropping the flows packets
+  * COMMANDS
+  *    update quantum: cannot not be done live
+  *    update Max Flow Queue Size: can be done live
+*/
+class DRR final : public Module {
+ public:
+    #define kDEFAULT_NUM_FLOWS 4096  // the default max number of flows allowed
```

Comment:
This is a mix of macro and constant style.  If you want to define these here, then use static const int kDefaultNumFlows, etc. instead of macro style.  If you want macro style, define them outside of the class block as DEFAULT_NUM_FLOWS, etc.